### DISCUSSION TITLE
BlueSCSI v1.1 doesn't work in Atari ST with ICD AdSCSI Micro ACSI-SCSI converter.

### DISCUSSION THREAD

**Discussion by M1kerochip** [erichelgeson/BlueSCSI-v1#268] [Q&A]
Hi,

I've got an Atari ST with an AdSCSI micro board from ICD.
(Or more accurately, it's a board that's been reverse engineered by t0ri, and as far as I know, is functionally an exact replica)

I've tested and used it with SCSI2SD v5.0 and the board works perfectly, so, I know the ST and the converter work ok.
(Transfer rate is benchmarking at around 960Kb/sec, and is perfectly fine with the SCSI2SD v5, with 3x HDD and one CDRom)

The ST boots up, but, the activity lights come on, and stay on, on the BlueSCSI, and once you get to the desktop, no HDD is accessible, and neither are either floppy (A:\ or B:\, connected separately, or together.)
(So, without the HDD booting, and floppy disabled, I can't load any software to detect the drive/adapter and partition etc)

I noticed there is a "MegaSTE" mode added in an earlier build where you can swap the ID and LUN, and I've tried that, but, it made no difference.
(The AdSCSI Micro only supports SCSI ID 0,1,2,3. It's designed for one single device)

The BlueSCSI v1.1 I built myself, but, I've tried it in an Amiga A2000 with GVP Zorro II card (SCSI+8mb fast) and, it's been working perfectly in that machine for months.
I even built a second BlueSCSI, and that too works in the GVP but not with the AdSCSI Micro.

The HDD Image I've tried is an 8GB image, one single HDD. Tried with 256MB, 512MB and so on, too.

The same image works fine on Hatari, and written to the SD Card on the SCSI2SD, not that it matters, since even a blank image fails to boot, and leaves the floppy disabled.

So, to that end, is there a debug build of the latest release?

**Comment by M1kerochip** [comment]
With the standard log.txt, you can see the BlueSCSI is starting fine, and mounting images etc.
This is the last config I tried, where I specifically named the HDD Images etc, but, it didn't work with just the defaults, either.
(I figured it was worth a shot)

[bluescsi.ini.txt](https://github.com/erichelgeson/BlueSCSI/files/15351947/bluescsi.ini.txt)
[LOG.txt](https://github.com/erichelgeson/BlueSCSI/files/15351948/LOG.txt)

**Comment by erichelgeson** [comment]
> So, to that end, is there a debug build of the latest release?

No, you can build one yourself buy editing DEBUG=1 in `BlueSCSI.cpp`

I've provided 2 builds here, both will send debug logs to serial out (use TerraTerm/Putty/CoolTerm to capture the logs) One for flashing via USB bootloader, one without.

[debug.zip](https://github.com/erichelgeson/BlueSCSI/files/15377661/debug.zip)

**Reply by M1kerochip** [reply]
Excellent, thanks! Hopefully I'll get some time to try it out shortly. If not, it'll be the weekend.

**Comment by M1kerochip** [comment]
OK.
Flashed, and it's not very helpful.
If I start it without an sd card, it says sd card not initialised.

Otherwise, as the ST boots up, I'm guessing I'm too slow connecting to the COM port to catch the config, but otherwise, all the serial output shows is:

CMD:

And nothing else. It's output just once.
